src/api/fetchData.js

```diff
@@ -2,24 +2,29 @@ import { useState, useEffect } from 'react';
 
 export default function fetchData(url) {
     const [data, setData] = useState([]);
-    const [loading, setLoading] = useState(null);
+    const [loading, setLoading] = useState(true);
     const [error, setError] = useState('');
 
+    const rawgioApiKey = ""
+    const requestOptions = {
+        headers: {
+            "Target-URL": "https://rawg.io",
+            "Authorization": "" // the cors-proxy expects this attribute, otherwise it fails
+        }
+    }
+
     useEffect(() => {
-        setLoading(true);
-        try {
-            fetch(url)
-                .then((res) => res.json())
-                .then((data) => setData(data));
-        } catch (e) {
-            throw new Error(
-                setError(
-                    `Cannot fetch the data, error with status ${response.status}`
+        fetch(`${url}?key=${rawgioApiKey}`, requestOptions)
+            .then((res) => res.json())
+            .then((data) => setData(data))
```

Comment:
- There should be an if that checks if the response is not ok, and throw a new error in case that's true.